include/tm/vector.hpp

```diff
@@ -100,21 +100,29 @@ class Vector {
     }
 
     void push_front(T val) {
+        insert(0, val);
+    }
+
+    void insert(size_t index, T val) {
         if (m_size >= m_capacity) {
             grow_at_least(m_size + 1);
         }
         m_size++;
-        for (size_t i = m_size - 1; i > 0; i--) {
+        for (size_t i = m_size - 1; i > index; i--) {
             m_data[i] = m_data[i - 1];
         }
-        m_data[0] = val;
+        m_data[index] = val;
     }
 
     T pop_front() {
-        assert(m_size != 0);
-        T val = m_data[0];
+        return pop_at(0);
+    }
+
+    T pop_at(size_t index) {
```

Comment:
I guess it's personal preference, but I've never seen a Vector `pop_at` method before and its use does seem limited. You're not using the "popped" value in your code anyway, so let's rename this to `remove()` and have it be a void return.

Reply:
I agree and it makes even more sense since I've ranamed it's counterpart to insert, thanks!